### PR TITLE
[Sources] Dear Systems

### PR DESCRIPTION
**Pull request by vellames**
Continuation of #6001 

### <samp>🤖 Generated by Copilot at 0b9d941</samp>

This pull request adds four new source components for the DEAR Inventory app that emit events based on different webhook types. The components are `available-stock-level-change`, `customer-updated`, `sale-shipment-tracking-number-change`, and `supplier-updated`. They all extend a common base component for webhooks and customize the event name and metadata.

### <samp>🤖 Generated by Copilot at 0b9d941</samp>

> _`DEAR Inventory`_
> _Webhook sources multiply_
> _Autumn harvest time_

### <samp>🤖 Generated by Copilot at 0b9d941</samp>

*  Add four new source components for the DEAR Inventory app that emit events based on different webhook types ([link](https://github.com/PipedreamHQ/pipedream/pull/6219/files?diff=unified&w=0#diff-a2d0f66b76b39b55eaa89dab760e0e9f6fca5f4ec988361c8a814161c3d7d14fR1-R29), [link](https://github.com/PipedreamHQ/pipedream/pull/6219/files?diff=unified&w=0#diff-bad7247ca747f662d633816fd34e84dd8f167cc43a9913c658bf7c18fda6cc9fR1-R25), [link](https://github.com/PipedreamHQ/pipedream/pull/6219/files?diff=unified&w=0#diff-110cc16e9abd01569f9327ed473157524615f194fe576d5e93592b81e961bc4fR1-R25), [link](https://github.com/PipedreamHQ/pipedream/pull/6219/files?diff=unified&w=0#diff-f6dd5022400c5cffb4b0470856e3876ee442dc1d4ff9c3206aeb3380ef2a683fR1-R25))

### CODE REVIEW

File: components/dear/sources/available-stock-level-change/available-stock-level-change.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+import { v4 as uuid } from "uuid";
+
+export default {
+  ...base,
+  name: "New Available Stock Level Change",
+  key: "dear-available-stock-level-change",
+  type: "source",
+  description: "Emit new event when the available stock level changes.",
```
Comment (alysonturing) [inline components/dear/sources/available-stock-level-change/available-stock-level-change.mjs:10]: Is it possible add the link to the documentation?

File: components/dear/sources/available-stock-level-change/available-stock-level-change.mjs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  version: "0.0.1",
+  methods: {
+    ...base.methods,
+    getWebhookType() {
+      return constants.WEBHOOK_TYPE.STOCK_AVAILABLE_STOCK_LEVEL_CHANGE;
+    },
+    getMetadata(payload) {
+      const id = uuid();
+      const item0 = payload[0];
+      return {
```
Comment (alysonturing) [inline components/dear/sources/available-stock-level-change/available-stock-level-change.mjs:20]: Does make sense to to check if the item is undefined?

File: components/dear/sources/customer-updated/customer-updated.mjs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+import { v4 as uuid } from "uuid";
+
+export default {
+  ...base,
+  name: "New Customer Updated",
+  key: "dear-customer-updated",
+  type: "source",
+  description: "Emit new event when a customer is updated.",
```
Comment (alysonturing) [inline components/dear/sources/customer-updated/customer-updated.mjs:10]: Is it possible add the link to the documentation?

File: components/dear/sources/sale-shipment-tracking-number-change/sale-shipment-tracking-number-change.mjs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+import { v4 as uuid } from "uuid";
+
+export default {
+  ...base,
+  name: "New Shipment Tracking Number Change",
+  key: "dear-sale-shipment-tracking-number-change",
+  type: "source",
+  description: "Emit new event when a shipment tracking number changes.",
```
Comment (alysonturing) [inline components/dear/sources/sale-shipment-tracking-number-change/sale-shipment-tracking-number-change.mjs:10]: Is it possible add the link to the documentation?

File: components/dear/sources/supplier-updated/supplier-updated.mjs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import constants from "../../common/constants.mjs";
+import base from "../common/webhooks.mjs";
+import { v4 as uuid } from "uuid";
+
+export default {
+  ...base,
+  name: "New Supplier Updated",
+  key: "dear-supplier-updated",
+  type: "source",
+  description: "Emit new event when a supplier is updated.",
```
Comment (alysonturing) [inline components/dear/sources/supplier-updated/supplier-updated.mjs:10]: Is it possible add the link to the documentation?

**Review by alysonturing — CHANGES_REQUESTED** [review]
Thank you for your valuable contribution! I added some comments hoping to improve it.

**Review by alysonturing — APPROVED** [review]
LGTM!

**Comment by michelle0927** [comment]
Testing passed, and ready for release!

Test report
[Dear_6129.pdf](https://github.com/PipedreamHQ/pipedream/files/11399685/Dear_6129.pdf)

**Pull request merged**